### PR TITLE
executor: refine the log msg when 'set character_set_results=null'

### PR DESCRIPTION
**Pull request by XuHuaiyu**
valStr is **only** used in the log message,
**NullDatum.ToString** will cause an error, 
before this commit,
we log this error message **as an error** which will confuse the users.

This commit log the error message **as a warning**,
and set a proper value for valStr when execute sql like `set character_set_results = null`.

PTAL @shenli @coocood

### CODE REVIEW

**Comment by coocood** [comment]
How about check if value is null first?

File: executor/set.go [hunk 1]
```diff
@@ -151,7 +150,13 @@ func (e *SetExecutor) executeSet() error {
 				return errors.Trace(err)
 			}
 			valStr, err := value.ToString()
-			terror.Log(errors.Trace(err))
+			if err != nil {
+				log.Warnf(errors.Trace(err).Error())
```
Comment (zimulala) [inline executor/set.go]: It's better to add some details to log.

**Comment by XuHuaiyu** [comment]
PTAL @coocood @zimulala

File: executor/set.go [hunk 2]
```diff
@@ -151,7 +150,11 @@ func (e *SetExecutor) executeSet() error {
 				return errors.Trace(err)
 			}
 			valStr, err := value.ToString()
-			terror.Log(errors.Trace(err))
+			if value.IsNull() {
```
Comment (coocood) [inline executor/set.go]: I mean check null before `ToString`.

**Comment by XuHuaiyu** [comment]
PTAL @coocood

File: executor/set.go [hunk 3]
```diff
@@ -150,8 +149,16 @@ func (e *SetExecutor) executeSet() error {
...
-			valStr, err := value.ToString()
-			terror.Log(errors.Trace(err))
+			var valStr string
+			if value.IsNull() {
+				valStr = "NULL"
+			} else {
+				var err error
+				valStr, err = value.ToString()
+				if err != nil {
+					log.Warnf("[CAN BE IGNORED]The value of set system variable cannot be converted to string, details: [%s]", errors.Trace(err).Error())
```
Comment (zimulala) [inline executor/set.go]: %v, errors.Trace(err)

**Comment by zimulala** [comment]
Rest LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**